### DISCUSSION TITLE
Add fan/ventilation speed support for Bosch compress 3800i/IVT Vent 402

### DISCUSSION THREAD

**Discussion by mattfro** [emsesp/EMS-ESP32#1690] [General]
Add fan control support. On this device I think you cannot turn off the fan and the min value is 20% and max 100%
I was trying to understand ems messages so(but I have no idea), if I guess correct, 40% should be 28 and 41% should be 29?
If that is the case, I think I fetched when I was changing the value in the pump between 40 and 41.
[K000+00:45:56.777 T 9959: [emsesp] controller(0x09) -W-> boiler(0x08), ?(0x04A5), data: 29 (offset 9)
[K000+00:46:01.127 T 9997: [emsesp] controller(0x09) -W-> boiler(0x08), ?(0x04A5), data: 28 (offset 9)
[K000+00:46:06.064 T 10064: [emsesp] controller(0x09) -W-> boiler(0x08), ?(0x04A5), data: 29 (offset 9)
[K000+00:46:08.889 T 10103: [emsesp] controller(0x09) -W-> boiler(0x08), ?(0x04A5), data: 28 (offset 9)
[K000+00:46:11.627 T 10146: [emsesp] controller(0x09) -W-> boiler(0x08), ?(0x04A5), data: 29 (offset 9)
**[K000+00:46:15.552 T 10191: [emsesp] controller(0x09) -W-> boiler(0x08), ?(0x04A5), data: 28 (offset 9)**
Last value I left it on was 40%

**Comment by proddy** [comment]
that sounds about right. Next make a write custom entity and see if it works! https://emsesp.github.io/docs/Configuring/#creating-custom-entities

**Comment by mattfro** [comment]
![image](https://github.com/emsesp/EMS-ESP32/assets/46936922/16984668-0ba2-4adb-ad45-9a67c114f161)
I hope this was correct, because it worked.

[K000+00:25:34.705 I 4107: [command] [0;33mCalling command 'custom/fan_speed' (set custom value on ems) with value 45
[K000+00:25:34.823 T 4108: [emsesp] Me(0x0B) -W-> boiler(0x08), ?(0x04A5), data: 2D (offset 9)

**Reply by proddy** [reply]
great, @MichaelDvP shall we add as a new device entity, or keep it as custom?

**Reply by MichaelDvP** [reply]
I think we should add. It's confirmed and i think usefull for all heatpump users.

**Comment by mattfro** [comment]
as I mentioned... for this pump, is only possible to set it between 20-100%. You can't turn it off, but useful for sure. For example summer time you might want more ventilation or if you have high humidity you also want more ventilate. Or there could be other reasons. :)

**Comment by MichaelDvP** [comment]
@mattfro Could you please do a `read 8 4A5` in terminal and post the complete telegram. It is unknown and it's always good to see example values and length if we search for other settings.
(Or use my dev-build, which fetches the telegram every minute, and post a log).

**Reply by mattfro** [reply]
here you go:
ems-esp:# read 8 4A5
000+12:19:44.023 N 53: [emsesp] boiler(0x08) -W-> Me(0x0B), ?(0x04A5), data: 00 00 3C 1D 09 0A 0A 01 00 28 0A 00 01 00 00
